### PR TITLE
Support Xcode report integration

### PR DESCRIPTION
**Pull request by elneruda**
Implement xcode report style : show warnings with line & column
#9 Suport multifile IBAction/IBOutlet extension

### CODE REVIEW

**Comment by fastred** [comment]
Thank you ❤️. This looks really nice! I'm going to take a closer look in a few days, when I have more time.

One significant suggestion that I have right now is to not include `bin/IBAnalyzer.app` in the repo. It's pretty large and would have to be updated with each release.

For now we could point people to [Releases](https://github.com/fastred/IBAnalyzer/releases) tab which contains prebuilt binaries. Later on I'm planning to distribute binary version through CocoaPods, similarly to [SwiftGen](https://github.com/SwiftGen/SwiftGen/blob/master/SwiftGen.podspec), and through Homebrew.

**Comment by elneruda** [comment]
@fastred great!  I've just improved severity, will add it in a new pull-request
noted for IBAnalyzer.app: will be removed from repo

**Comment by fastred** [comment]
Cool. Would be great if you could remove `IBAnalyzer.app` in this PR. I'd squash it before merging, so that it wouldn't increase the repo's size.

**Comment by elneruda** [comment]
@fastred I've just removed  IBPAnalyzer.app

File: IBAnalyzer.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -366,7 +366,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "if which ${PODS_ROOT}/SwiftLint/swiftlint >/dev/null; then\ncd IBAnalyzer && ${PODS_ROOT}/SwiftLint/swiftlint && cd ../IBAnalyzerTests && ${PODS_ROOT}/SwiftLint/swiftlint && cd ..\nelse\necho \"warning: SwiftLint not installed, run 'pod install'\"\nfi\n";
+			shellScript = "if which ${PODS_ROOT}/SwiftLint/swiftlint >/dev/null; then\n${PODS_ROOT}/SwiftLint/swiftlint\nelse\necho \"warning: SwiftLint not installed, run 'pod install'\"\nfi\n";
```
Comment (fastred) [inline IBAnalyzer.xcodeproj/project.pbxproj]: This is a lot nicer than what we had before. Thanks!

File: IBAnalyzer/Analyzers/ConnectionAnalyzer.swift [hunk 2]
```diff
@@ -7,25 +7,127 @@
...
+        self.column = column
+        self.url = url
+        self.isOptional = isOptional
+    }
+
+    init(name: String, file: File, offset: Int64, isOptional: Bool = false) {
+        let fileOffset = Violation.getLineColumnNumber(of: file, offset: Int(offset))
+        var url: URL?
+        if let path = file.path {
+            url = URL(string: path)
```
Comment (fastred) [inline IBAnalyzer/Analyzers/ConnectionAnalyzer.swift]: Better to use `URL(fileURLWithPath:)` as `URL(string:)` sometimes returns `nil` for paths.

File: IBAnalyzer/Analyzers/ConnectionAnalyzer.swift [hunk 3]
```diff
@@ -7,25 +7,127 @@
 //
 
 import Foundation
+import SourceKittenFramework
+
+struct Violation {
```
Comment (fastred) [inline IBAnalyzer/Analyzers/ConnectionAnalyzer.swift]: I like how this structure encapsulates some additional data needed for better warning/error reporting but I think its name should be improved. At the time of parsing we only know that it represents an action or outlet but we aren't yet sure if it'll be a violation.

I'd consider naming it `Declaration` instead.

File: IBAnalyzer/Analyzers/ConnectionAnalyzer.swift [hunk 4]
```diff
@@ -7,25 +7,127 @@
...
+        if let column = lines.last?.characters.count {
+            return (line: lines.count, column: column)
+        }
+        return (line: lines.count, column: 0)
+    }
+}
+
+extension Violation: Equatable {
+    public static func == (lhs: Violation, rhs: Violation) -> Bool {
+        return lhs.name == rhs.name
```
Comment (fastred) [inline IBAnalyzer/Analyzers/ConnectionAnalyzer.swift]: Shouldn't we also compare values of other properties?

File: IBAnalyzer/Analyzers/ConnectionAnalyzer.swift [hunk 5]
```diff
@@ -7,25 +7,127 @@
...
         case let .UnnecessaryAction(className: className, action: action):
-            return "\(className) contains unused @IBAction named: \(action)"
+            return "\(action.description): warning: IBAction unused: \(action.name) not linked in \(action.fileName(className: className))"
+        }
+    }
+
+    var isSeriousViolation: Bool {
+        switch self {
+        case .MissingOutlet(className: _, outlet: _), .MissingAction(className: _, action: _):
+            return true
```
Comment (fastred) [inline IBAnalyzer/Analyzers/ConnectionAnalyzer.swift]: Good idea!

File: IBAnalyzerTests/Stubs.swift [hunk 6]
```diff
@@ -41,21 +45,31 @@ struct StubNibParser: NibParserType {
...
+    static let buttonTapped = Violation(name: "buttonTapped:", line: 1, column: 0)
+    static let eMap = ["FirstViewController": Class(outlets: [], actions: [StubSwiftParser.buttonTapped], inherited: [])]
 
-    func mappingForFile(at url: URL) throws -> [String: Class] {
+    func mappingForFile(at url: URL, result: inout [String: Class]) throws {
         switch url {
         case URL(fileURLWithPath: "a.swift"):
-            return type(of: self).aMap
+            result += type(of: self).aMap
+            //return type(of: self).aMap
```
Comment (fastred) [inline IBAnalyzerTests/Stubs.swift]: Let's remove the commented out code.

File: IBAnalyzerTests/Stubs.swift [hunk 7]
```diff
@@ -41,21 +45,31 @@ struct StubNibParser: NibParserType {
...
         case URL(fileURLWithPath: "e.swift"):
-            return type(of: self).eMap
+            result += type(of: self).eMap
+            //return type(of: self).eMap
```
Comment (fastred) [inline IBAnalyzerTests/Stubs.swift]: Let's remove the commented out code.

File: IBAnalyzer/Analyzers/ConnectionAnalyzer.swift (hunk 3 above)
Comment (elneruda) [inline IBAnalyzer/Analyzers/ConnectionAnalyzer.swift]: agreed for naming

**Comment by elneruda** [comment]
@fastred i've just pushed requested improvements

File: IBAnalyzer/Analyzers/ConnectionAnalyzer.swift (hunk 2 above)
Comment (elneruda) [inline IBAnalyzer/Analyzers/ConnectionAnalyzer.swift]: ok sure

File: IBAnalyzer/Analyzers/ConnectionAnalyzer.swift (hunk 4 above)
Comment (elneruda) [inline IBAnalyzer/Analyzers/ConnectionAnalyzer.swift]: i'm not sure, optional appears only from source code side and not from xib

**Comment by fastred** [comment]
Thanks! I'll get the CocoaPods integration sorted out soon.

**Comment by elneruda** [comment]
great!!

**Pull request merged**